lib/cosmos/dart/script/dart_decom_server.rb

```diff
@@ -0,0 +1,201 @@
+require File.expand_path('../../config/environment', __FILE__)
+require 'dart_common'
+require 'dart_logging'
+
+# Rails Json screws up COSMOS handling of Nan, etc.
+require "active_support/core_ext/object/json"
+module ActiveSupport
+  module ToJsonWithActiveSupportEncoder # :nodoc:
+    def to_json(options = nil)
+      super(options)
+    end
+  end
+end
+
+class DartDecomServer
+  include DartCommon
+
+  def query(request)
+    request_start_time = Time.now
+    Cosmos::Logger.info("#{request_start_time.formatted}: QUERY: #{request}")
+
+    begin
+      start_time_sec = request['start_time_sec']
+      start_time_usec = request['start_time_usec']
+      end_time_sec = request['end_time_sec']
+      end_time_usec = request['end_time_usec']
+
+      start_time = nil
+      end_time = nil
+      start_time = Time.at(start_time_sec, start_time_usec) if start_time_sec and start_time_usec
```

Comment:
Should this be Time.at(sec, usec).sys? @donaldatball changed this throughout the code a while back.

Reply:
I believe .sys only affects timestamps that are displayed, so I don't think so.

Reply:
It converts the timestamp to UTC or not depending on whether TIME_ZONE_UTC is set in system.txt. If the users have this set and request time using UTC this code will currently return local time.

Reply:
The sec/usec values should always be UTC when sent on the line.  If not that is a bug.